### PR TITLE
ENT-1783: Replace edx_rbac.utils.get_decoded_jwt_from_request with edx_rest_framework_extensions.auth.jwt.cookies.get_decoded_jwt

### PR DESCRIPTION
**Pull request by saleem-latif**
**Description:** 
This PR removes the usages of `get_decoded_jwt_from_request` from rbac in favor of `get_decoded_jwt` from edx-drf-extensions. It also adds `get_request_or_stub` and updates the usages.

### CODE REVIEW

File: ecommerce/enterprise/utils.py [hunk 1]
```diff
@@ -481,3 +483,32 @@ def get_enterprise_id_for_user(site, user):
         pass
 
     return None
+
+
+def get_request_or_stub():
```
Comment (mushtaqak) [inline ecommerce/enterprise/utils.py]: Are we planning to remove rbac get_[request_or_stub](https://github.com/edx/edx-rbac/blob/master/edx_rbac/utils.py#L21) and get_decoded_jwt_from_request  methods after these PRs are merged ?

**Review by mushtaqak — APPROVED** [review]
Looks good to me 👍

File: ecommerce/enterprise/utils.py [hunk 2]
```diff
@@ -481,3 +483,32 @@ def get_enterprise_id_for_user(site, user):
         pass
 
     return None
+
+
```
Comment (mushtaqak) [inline ecommerce/enterprise/utils.py]: I think we should mention taken from edx-platform ?

File: ecommerce/enterprise/utils.py (hunk 1 above)
Comment (brittneyexline) [reply ecommerce/enterprise/utils.py]: as mentioned in the other PRs, i think we should stop using this function and call crum directly, and adjust the tests as needed.

**Comment by saleem-latif** [comment]
@brittneyexline PR is ready for another review.

**Review by brittneyexline — APPROVED** [review]

**Comment by brittneyexline** [comment]
@saleem-latif - can you please squash your commits before i merge this?

**Comment by saleem-latif** [comment]
@brittneyexline I have squashed the commits, PR is ready for merge.

**Comment by brittneyexline** [comment]
@saleem-latif looks like we have some conflicts with the requirements files that need to be resolved.

**Comment by saleem-latif** [comment]
@brittneyexline rebased, it is ready for merge

**Comment by brittneyexline** [comment]
@saleem-latif - i had to revert your pr, see the description there for more details: https://github.com/edx/ecommerce/pull/2253

_i think the best path forward is to figure out what authentication ecommerce_worker is using and possibly do a check here (and wherever else we call `get_decoded_jwt_from_jwt_cookie`) to catch invalid token errors: https://github.com/edx/ecommerce/blob/master/ecommerce/core/models.py#L547_

UPDATE: Ignore the above proposal - see comments in the ticket!

**Pull request merged**